core/src/broadcast_stage/fail_entry_verification_broadcast_run.rs

```diff
@@ -31,44 +40,77 @@ impl BroadcastRun for FailEntryVerificationBroadcastRun {
         let bank = receive_results.bank.clone();
         let last_tick_height = receive_results.last_tick_height;
 
-        // 2) Convert entries to shreds + generate coding shreds. Set a garbage PoH on the last entry
-        // in the slot to make verification fail on validators
-        if last_tick_height == bank.max_tick_height() {
-            let mut last_entry = receive_results.entries.last_mut().unwrap();
-            last_entry.hash = Hash::default();
+        if bank.slot() != self.current_slot {
+            self.next_shred_index = 0;
+            self.current_slot = bank.slot();
         }
 
-        let next_shred_index = blockstore
-            .meta(bank.slot())
-            .expect("Database error")
-            .map(|meta| meta.consumed)
-            .unwrap_or(0) as u32;
+        // 2) If we're past SLOT_TO_RESOLVE, insert the correct shreds so validators can repair
+        // and make progress
+        if bank.slot() > SLOT_TO_RESOLVE && !self.good_shreds.is_empty() {
+            info!("Resolving bad shreds");
+            let mut shreds = vec![];
+            std::mem::swap(&mut shreds, &mut self.good_shreds);
+            blockstore_sender.send((Arc::new(shreds), None))?;
+        }
+
+        // 3) Convert entries to shreds + generate coding shreds. Set a garbage PoH on the last entry
+        // in the slot to make verification fail on validators
+        let last_entries = {
+            if last_tick_height == bank.max_tick_height() && bank.slot() < NUM_BAD_SLOTS {
```

Comment:
qq: in a similar spirit of https://github.com/solana-labs/solana/pull/9805/files/70a2dd47c0cb8e7d9d20d4d29928643115b5b880#r418466061, is `bank.slot() < NUM_BAD_SLOTS` right?

Reply:
same thing as above, test harness code